gradio/components/login_button.py

```diff
@@ -0,0 +1,88 @@
+"""Predefined button to sign in with Hugging Face in a Gradio Space."""
+from __future__ import annotations
+
+from typing import Any, Literal
+
+from gradio_client.documentation import document, set_documentation_group
+
+from gradio.components import Button
+from gradio.routes import Request
+
+set_documentation_group("component")
+
+
+@document()
+class LoginButton(Button):
```

Comment:
Should we call these `HFLoginButton` and similarly for logout? I suspect there will be interest in this functionality for other auth providers and even the vanilla gradio auth

Reply:
I think we can add a parameter in `LoginButton` or `LogoutButton` specifying the authentication provider -- later once we have support for other auths.

Reply:
Although we do have the vanilla gradio auth at the moment, good point